src/terrain/services/filesystem/directory.clj

```diff
@@ -162,7 +164,12 @@
 (defn do-paged-listing
   "Entrypoint for the API that calls (paged-dir-listing)."
   [{user :shortUsername} {:keys [path] :as params}]
-  (let [params (dissoc params :path)]
-    (->> (fix-paged-listing-params user params)
-         (data/list-folder-contents path)
-         (format-page user))))
+  (if-let [listing-user (if (= path (cfg/fs-community-data))
+                          (or user "anonymous")
+                          user)]
+    (let [params (dissoc params :path)]
+      (->> (fix-paged-listing-params listing-user params)
+           (data/list-folder-contents path)
+           (format-page user)
+           (ok)))
+    (response/unauthorized "No authentication information found in the request")))
```

Comment:
I'd keep the `ok` up at the `route` level, since I think of the `ok` and `unauthorized` responses as specific to the routes (imagine if we had to switch from `compojure-api` to something else one of these days).

Our compojure-api routes have exception handling that will catch exceptions and return the appropriate error response. For example, take a look at `terrain.services.filesystem.validators/not-superuser`
https://github.com/cyverse-de/terrain/blob/8cfae8a1f23a3f23d5ae90f12b3c92f03af1061b/src/terrain/services/filesystem/validators.clj#L12-L16
It throws a `clojure-commons.exception/not-authorized` exception type, which gets handled by `clojure-commons.exception/not-authorized-handler`:
https://github.com/cyverse-de/iplant-clojure-commons/blob/99c9d70cbe77be66b3c67a467de0fc887fe93359/src/clojure_commons/exception.clj#L113-L114

So the `ok` could be kept back at the route level, and this function could throw an exception in the `else` part of this `if-let`.

Reply:
Ah ok, thanks for providing an example!  I'll try this out and refactor